### PR TITLE
feat: add perf-flamegraph tools in docker image

### PR DESCRIPTION
**Pull request by huangjw806**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Install perf flamegraph in docker image

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

### CODE REVIEW

**Review by lmatz — APPROVED** [review]
LGTM

File: docker/Dockerfile [hunk 1]
```diff
@@ -35,7 +35,7 @@ RUN rustup self update \
   && rustup show \
   && rustup component add rustfmt
 
-
+RUN cargo install flamegraph
```
Comment (xxchan) [inline docker/Dockerfile:38]: Ideally we should prefer install binary directly, but it seems not very slow to compile, so acceptable to me.

File: docker/Dockerfile [hunk 2]
```diff
@@ -115,6 +115,9 @@ FROM base AS risingwave
 
 LABEL org.opencontainers.image.source https://github.com/risingwavelabs/risingwave
 
+RUN apt-get update && apt-get -y install linux-tools-generic \
+    && ln -s "$(find /usr/lib/linux-tools/*/perf | head -1)" /usr/local/bin/perf
```
Comment (xxchan) [inline docker/Dockerfile:119]: nit: no need to `apt-update` any more as we did it at the very beginning.
```suggestion
RUN apt-get -y install linux-tools-generic \
    && ln -s "$(find /usr/lib/linux-tools/*/perf | head -1)" /usr/local/bin/perf
```

**Review by xxchan — APPROVED** [review]

File: docker/Dockerfile (hunk 2 above)
Comment (huangjw806) [reply docker/Dockerfile:119]: I thought so at first, but it doesn’t seem to work.
https://buildkite.com/risingwavelabs/docker/builds/14698#018bfaf6-f601-4056-8f8c-f22186a8d5b7

**Review by fuyufjh — APPROVED** [review]

**Review by MrCroxx — APPROVED** [review]
LGTM!

BTW, I used the `(cargo-)flamegraph` repo just because I am familiar with it. This repository internally uses linux perf to generate flame graphs. If the overhead of installing this repository is too large, you can use https://github.com/brendangregg/FlameGraph .

**Pull request merged**